modules/setting/storage.go

```diff
@@ -158,10 +158,26 @@ func getStorage(rootCfg ConfigProvider, name, typ string, sec ConfigSection) (*S
 
 	switch targetType {
 	case string(LocalStorageType):
-		storage.Path = ConfigSectionKeyString(targetSec, "PATH", filepath.Join(AppDataPath, name))
+		targetPath := ConfigSectionKeyString(targetSec, "PATH", "")
+		if targetPath == "" {
+			targetPath = AppDataPath
+		}
+
+		extraConfigSec := sec
+		if extraConfigSec == nil {
+			extraConfigSec = storageNameSec
+		}
+
+		if extraConfigSec == nil {
+			storage.Path = filepath.Join(targetPath, name)
+		} else {
+			storage.Path = ConfigSectionKeyString(extraConfigSec, "PATH", filepath.Join(targetPath, name))
+		}
+
 		if !filepath.IsAbs(storage.Path) {
 			storage.Path = filepath.Join(AppWorkPath, storage.Path)
```

Comment:
It should report an error (not abs path). This should be an impossible case in real production.

Reply:
`ConfigSectionKeyString(extraConfigSec, "PATH",filepath.Join(targetPath, name)` may return an relative path?

Reply:
The question is "why".

For consistent, the "targetPath" itself should be absolute first.

Reply:
Changed to `AppDataPath`